types/src/system/auction/providers.rs

```diff
@@ -93,6 +93,16 @@ pub trait MintProvider {
     /// Reduce total supply by `amount`. Returns unit on success, otherwise
     /// an error.
     fn reduce_total_supply(&mut self, amount: U512) -> Result<(), Error>;
+
+    /// Transfers token from a given source URef to a bonding purse.
+    /// Returns the number of tokens currently delegated to a given validator.
+    fn handle_delegation(
```

Comment:
I don't think `Mint` should be coupled with auction-like logic (`delegation` is specific to auction and has nothing to do with minting/destroying/transferring tokens)

Reply:
Well, the handling of delegation requires that token be transferred to the necessary bonding purses and so I feel that its okay for it to be here

Reply:
This method could be implemented on an `Auction` trait directly:
```rust
pub trait Auction:
    StorageProvider + RuntimeProvider + MintProvider + AccountProvider + Sized
{
```
which means that implementations of methods in that trait will have access to other methods available on `Mint*` and `Runtime*` and `Storage*`. That should be enough to implement correct logic and not burden `Mint` with auction-related logic and concepts.

Reply:
Removed from the trait and used @mpapierski of putting it in `detail.rs`